README.md

```diff
@@ -186,6 +186,53 @@ variable_name:
 reporter: "csv" # reporter type (xcode, json, csv, checkstyle)
 ```
 
+#### Defining Custom Rules
+
+You can define custom regex-based rules in you configuration file using the
+following syntax:
+
+```yaml
+custom_rules:
+  pirates_beat_ninjas: # rule identifier
+    name: "Pirates Beat Ninjas" # rule name. optional.
+    regex: "([n,N]inja)" # matching pattern
+    match_kinds: # SyntaxKinds to match. optional.
+      - comment
+      - identifier
+    message: "Pirates are better than ninjas." # violation message. optional.
+    severity: error # violation severity. optional.
+  no_hiding_in_strings:
+    regex: "([n,N]inja)"
+    match_kinds: string
+```
+
+This is what the output would look like:
+
+![](custom-rule.png)
+
+By default, custom rules will match against all syntax kinds. You can filter
```

Comment:
I still don't think this sufficiently conveys the logic used to match syntax kinds. Consider making this clearer, such as:

> You can filter the matches by providing one or more `match_kinds`, which will reject matches that include syntax kinds that are not present in this list. Here are all the possible syntax kinds: